### ISSUE TITLE
ExclusiveStartKey returns result set including the element with the start key

### ISSUE THREAD

**Issue by mantoni** [mhart/dynalite#95]
Hey, thanks for making this, it's help a lot with testing my app!

I came accross a difference to DynamoDB when specifying `ExclusiveStartKey`. Dynalite returns the the element with the specified key while DynamoDB doesn't. This means when I use a query results `LastEvaluatedKey` as the starting point for the next query I end up with duplicate entries.

Dynalite v2.0.0
Node v10.6.0
macOS High Sierra

**Comment by mhart** [comment]
@mantoni can you show me an example query where this happens? Thanks!

**Comment by mantoni** [comment]
I manged to create a runnable test case that reproduces the issue. It turns out to be related to `ScanIndexForward: false`.

```js
'use strict';

const AWS = require('aws-sdk');
const dynalite = require('dynalite');

const config = {
  region: 'us-east-1',
  endpoint: 'http://localhost:8001',
  accessKeyId: 'akid',
  secretAccessKey: 'secret',
  apiVersion: '2012-08-10'
};

function then(cb) {
  return (err, res) => {
    if (err) {
      throw err;
    }
    cb(res);
  };
}

function createTable(done) {
  const db = new AWS.DynamoDB(config);
  db.createTable({
    TableName: 'Test',
    KeySchema: [{
      AttributeName: 'account',
      KeyType: 'HASH'
    }, {
      AttributeName: 'num',
      KeyType: 'RANGE'
    }],
    AttributeDefinitions: [{
      AttributeName: 'account',
      AttributeType: 'S'
    }, {
      AttributeName: 'num',
      AttributeType: 'N'
    }],
    ProvisionedThroughput: {
      ReadCapacityUnits: 1,
      WriteCapacityUnits: 1
    }
  }, then(done));
}

const client = new AWS.DynamoDB.DocumentClient(config);

function fillTable(done) {
  client.put({
    TableName: 'Test',
    Item: { account: 'abc', num: 1 }
  }, then(() => {

    client.put({
      TableName: 'Test',
      Item: { account: 'abc', num: 2 }
    }, then(done));
  }));
}

function runTest(done) {
  client.query({
    TableName: 'Test',
    ScanIndexForward: false,
    KeyConditionExpression: 'account=:a',
    ExpressionAttributeValues: { ':a': 'abc' },
    Limit: 1
  }, then(doc => {

    console.log('#1', doc.Items);

    client.query({
      TableName: 'Test',
      ScanIndexForward: false,
      KeyConditionExpression: 'account=:a',
      ExpressionAttributeValues: { ':a': 'abc' },
      ExclusiveStartKey: doc.LastEvaluatedKey
    }, then(doc => {

      console.log('#2', doc.Items);

      done();
    }));
  }));
}

const dynamo = dynalite({
  createTableMs: 1,
  updateTableMs: 1
}).listen(8001, then(() => {

  createTable(() => {
    fillTable(() => {
      runTest(() => {
        dynamo.close();
      });
    });

  });
}));
```

This prints:

```
#1 [ { account: 'abc', num: 2 } ]
#2 [ { account: 'abc', num: 2 }, { account: 'abc', num: 1 } ]
```

I don't expect it to return the entry with `num: 2` on the second query. When I remove `ScanIndexForward: false` it works as expected.

**Comment by mhart** [comment]
Awesome! Thanks for the repro – I think you're right that I don't have enough tests with `ScanIndexForward: false` and `ExclusiveStartKey` – I'll look into it today

**Comment by mhart** [comment]
Fixed in v2.1.0

**Comment by mantoni** [comment]
Awesome! Thank you 🙏
